src/pymor/algorithms/rules.py

```diff
@@ -382,7 +391,9 @@ def get_children(cls, obj):
                 v = getattr(obj, k)
                 if (isinstance(v, Operator)
                         or isinstance(v, dict) and all(isinstance(vv, Operator) or vv is None for vv in v.values())
-                        or isinstance(v, (list, tuple)) and all(isinstance(vv, Operator) or vv is None for vv in v)):
+                        or isinstance(v, (list, tuple)) and all(isinstance(vv, Operator) or vv is None for vv in v)
+                        or isinstance(v, np.ndarray) and v.dtype == object and v.ndim >= 1
+                            and all(isinstance(vv, Operator) or vv is None for vv in v.flat)):
```

Comment:
Hmm, there is an interesting corner case. What should happen with an array that only contains `None`?

Is it more likely that this is an attribute of children that happens to contain no children at all, or is it some arbitrary NumPy array of `None`s, which could never contain an `Operator`? It doesn't matter too much, since `apply_children` wouldn't do anything, anyway.

I'd tend to exclude these attributes as children as the less surprising behaviour. In that case, one should also adapt the conditions for dict, list and tuple. What do you thin?